### ISSUE TITLE
Data commitment roots for empty blocks returning same result

### ISSUE THREAD

**Issue by SweeXordious** [celestiaorg/celestia-core#733]
As per the change introduced in https://github.com/celestiaorg/celestia-core/pull/625, we have data commitment being generated from the data roots of a set of blocks. 
The problem is that for empty blocks, we end up having the same the commitment over the range. This creates an issue when signing data commitments by the QGB orchestrator, as it finds the same commitment and thinks it is a duplicate signature.
One possible solution is to prepend the generated data commitment with the range.

**Comment by SweeXordious** [comment]
@evan-forbes What do you think ?

**Comment by adlerjohn** [comment]
https://github.com/celestiaorg/quantum-gravity-bridge/blob/c61578fc1d05b9a9c607a94ffcfef5424be7ce33/src/QuantumGravityBridge.sol#L195

**Comment by SweeXordious** [comment]
Yes, but this is when relaying attestations.
The problem is with the `_dataRootTupleRoot` when signing it. That one is the same for ranges of empty blocks.

**Comment by SweeXordious** [comment]
Another solution would be to make `(commitment, orchestrator_address, begin_block, end_block)` the key in the state db instead of only `(commitment, orchestrator_address)`

**Comment by adlerjohn** [comment]
> The problem is with the _dataRootTupleRoot when signing it.

But the `_dataRootTupleRoot` isn't what's signed? What's signed is the hash `keccak256(abi.encode(_bridge_id, DATA_ROOT_TUPLE_ROOT_DOMAIN_SEPARATOR, _oldNonce, _newNonce, _dataRootTupleRoot))`.

**Comment by SweeXordious** [comment]
Yes, sorry. 
What I meant was that Celestia is storing the signatures using the following key: `(commitment, orchestrator_address)`, with the commitment being what the orchestrator receives from tendermint.
The issue, is that the commitment received from tendermint is the same. Thus, we end up with a duplicate key in the db: `(commitment, orchestrator_address)`.

**Comment by evan-forbes** [comment]
could we do (commitment, nonce, orchestrator_address)?

**Comment by SweeXordious** [comment]
True, for `(commitment, nonce, orchestrator_address)`, what is the nonce ? `endBlock % window` ?
